contrib/parseq-examples/src/main/java/com/linkedin/parseq/example/batching/BatchingClientExample.java

```diff
@@ -0,0 +1,70 @@
+/* $Id$ */
+package com.linkedin.parseq.example.batching;
+
+import java.util.List;
+
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.batching.BatchEntry;
+import com.linkedin.parseq.batching.BatchingStrategy;
+import com.linkedin.parseq.batching.BatchingSupport;
+import com.linkedin.parseq.example.common.AbstractExample;
+import com.linkedin.parseq.example.common.ExampleUtil;
+import com.linkedin.parseq.promise.Promises;
+
+
+/**
+ * @author Jaroslaw Odzga (jodzga@linkedin.com)
+ */
+public class BatchingClientExample extends AbstractExample {
+
+  public static class ExampleBatchingStrategy extends BatchingStrategy<Long, Long> {
+
+    @Override
+    public Long groupForKey(Long key) {
+      //lets say all tasks can be grouped into one batch
+      return 0L;
+    }
+
+    @Override
+    public Task<?> taskForBatch(final Long group, final List<? extends BatchEntry<Long, Object>> batch) {
+      return Task.action("batchExecute", () -> {
+        System.out.println("batch: " + batch);
+        batch.forEach(entry -> entry.getPromise().done("value for id + " + entry.getKey()));
+      });
+    }
+  }
+
+  final ExampleBatchingStrategy batchingStrategy = new ExampleBatchingStrategy();
+
+  Task<String> batchableTask(final Long id) {
+    return batchingStrategy.batchable("fetch id: " + id, () -> {
+      return new BatchEntry<Long, String>(id, Promises.settable());
+    });
+  }
+
+  @Override
+  protected void customizeEngine(com.linkedin.parseq.EngineBuilder engineBuilder) {
+    BatchingSupport batchingSupport = new BatchingSupport();
+    batchingSupport.registerStrategy(batchingStrategy);
```

Comment:
So you'll be able to register multiple different things that can be batched and we'll figure out the correct one based on the plan context? Have I understood this correctly?